Can't start on Proxmox/Pi5

Hi - looking to try this out, followed the Proxmox LXC instructions on a working Proxmox PXVIRT 9.0 on a Pi5 and got - 

[OK] Container created
[INFO] Starting container...
explicitly configured lxc.apparmor.profile overrides the following settings: features:nesting
sync_wait: 34 An error occurred in another process (expected sequence number 7)
__lxc_start: 2119 Failed to spawn container "106"
startup for container '106' failed

[ERROR] Installation failed. Check the output above for errors.

Thanks for checking it out, but still no joy here. I will find another machine to try your code out on, but in the mean time - in case you can spot something - here is the entire stdout from the install (still can't find anything more useful):

```

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  Confirm Settings
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

Container Settings:
  ID:        107
  Hostname:  network-optimizer
  Debian:    13
  RAM:       2048MB
  Swap:      512MB
  CPU:       2 cores
  Disk:      10GB
  Storage:   local
  Bridge:    vmbr0
  VLAN Tag:  none (untagged)
  IP:        dhcp
  SSH:       disabled

Application Settings:
  Timezone:       Europe/London
  Web UI Port:    8042 (fixed)
  Speedtest Port: 3005
  iperf3 Server:  disabled
  Host Redirect:  disabled
  Reverse Proxy:  none
  Geo Location:   disabled
  Password:       (auto-generate)

Proceed with installation? [Y/n]: 

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  Downloading Container Template
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

[INFO] Finding Debian 13 template...
[OK] Found template: debian-13-standard_13.1-2_amd64.tar.zst
[OK] Template already downloaded

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  Creating LXC Container
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

[INFO] Creating container 107 (network-optimizer)...
Formatting '/var/lib/vz/images/107/vm-107-disk-0.raw', fmt=raw size=10737418240 preallocation=off
Creating filesystem with 2621440 4k blocks and 655360 inodes
Filesystem UUID: 20b465cc-6a8b-4c45-b862-3c6203fc4fd3
Superblock backups stored on blocks: 
        32768, 98304, 163840, 229376, 294912, 819200, 884736, 1605632
extracting archive '/var/lib/vz/template/cache/debian-13-standard_13.1-2_amd64.tar.zst'
Total bytes read: 551505920 (526MiB, 273MiB/s)
Detected container architecture: amd64
Creating SSH host key 'ssh_host_ecdsa_key' - this may take some time ...
done: SHA256:L/chyFPXVYfl26e1vHvKPkcDTFf/dn5NXpPWrs5Wq1A root@network-optimizer
Creating SSH host key 'ssh_host_ed25519_key' - this may take some time ...
done: SHA256:v1U0Vb1/kkjVyOI7epRjGHDKAtAAZV5qEzDZlpEXyrQ root@network-optimizer
Creating SSH host key 'ssh_host_rsa_key' - this may take some time ...
done: SHA256:JwwmR/KH34hR0MG4HNLh7ycpy5kVbzNjyTPMUv7Tdlw root@network-optimizer
[OK] Container created
[INFO] Starting container...
sync_wait: 34 An error occurred in another process (expected sequence number 7)
__lxc_start: 2119 Failed to spawn container "107"
startup for container '107' failed

[ERROR] Installation failed. Check the output above for errors.
To clean up the failed container:
  pct stop 107 2>/dev/null; pct destroy 107
```

--- a/scripts/proxmox/install.sh
+++ b/scripts/proxmox/install.sh
@@ -187,12 +187,31 @@ find_debian_template() {
         aarch64) host_arch="arm64" ;;
     esac
 
-    # Find the latest debian template for the selected version and architecture
+    local available
+    available=$(pveam available -section system 2>/dev/null | awk '{print $2}')
+
+    # Find the latest debian template for the selected version and architecture.
+    # Standard PVE naming: debian-13-standard_13.1-2_amd64.tar.zst
     local template
-    template=$(pveam available -section system 2>/dev/null | awk '{print $2}' | grep "debian-${version}-standard" | grep "_${host_arch}\.tar" | tail -1 || true)
+    template=$(echo "$available" | grep "debian-${version}-standard" | grep "_${host_arch}\.tar" | tail -1 || true)
+
+    # ARM ports of Proxmox (PXVIRT) publish their own templates named by
+    # codename and build date instead: debian-trixie-20260328_arm64.tar.xz
+    if [[ -z "$template" ]]; then
+        local codename=""
+        case "$version" in
+            11) codename="bullseye" ;;
+            12) codename="bookworm" ;;
+            13) codename="trixie" ;;
+            14) codename="forky" ;;
+        esac
+        if [[ -n "$codename" ]]; then
+            template=$(echo "$available" | grep "^debian-${codename}-" | grep "_${host_arch}\.tar" | tail -1 || true)
+        fi
+    fi
 
     if [[ -z "$template" ]]; then
-        msg_error "Could not find Debian ${version} template for ${host_arch} in repository."
+        msg_error "Could not find a Debian ${version} template for ${host_arch} in the repository."
         msg_info "Available templates:"
         pveam available -section system 2>/dev/null | grep -i debian | head -5
         exit 1

--- a/scripts/proxmox/install.sh
+++ b/scripts/proxmox/install.sh
@@ -177,12 +177,22 @@ find_debian_template() {
     # Update template list
     pveam update &>/dev/null || true
 
-    # Find the latest debian template for the selected version
+    # Match the host architecture. ARM ports of Proxmox (e.g. PXVIRT on a
+    # Raspberry Pi) can list templates for multiple architectures, and an
+    # amd64 rootfs fails to start on an arm64 host.
+    local host_arch
+    host_arch=$(dpkg --print-architecture 2>/dev/null || uname -m)
+    case "$host_arch" in
+        x86_64) host_arch="amd64" ;;
+        aarch64) host_arch="arm64" ;;
+    esac
+
+    # Find the latest debian template for the selected version and architecture
     local template
-    template=$(pveam available -section system 2>/dev/null | grep "debian-${version}-standard" | tail -1 | awk '{print $2}')
+    template=$(pveam available -section system 2>/dev/null | awk '{print $2}' | grep "debian-${version}-standard" | grep "_${host_arch}\.tar" | tail -1 || true)
 
     if [[ -z "$template" ]]; then
-        msg_error "Could not find Debian ${version} template in repository."
+        msg_error "Could not find Debian ${version} template for ${host_arch} in repository."
         msg_info "Available templates:"
         pveam available -section system 2>/dev/null | grep -i debian | head -5
         exit 1

--- a/scripts/proxmox/install.sh
+++ b/scripts/proxmox/install.sh
@@ -211,9 +211,13 @@ find_debian_template() {
     fi
 
     if [[ -z "$template" ]]; then
-        msg_error "Could not find a Debian ${version} template for ${host_arch} in the repository."
-        msg_info "Available templates:"
-        pveam available -section system 2>/dev/null | grep -i debian | head -5
+        # stdout is captured by the caller's command substitution, so the
+        # error must go to stderr to be visible
+        {
+            msg_error "Could not find a Debian ${version} template for ${host_arch} in the repository."
+            msg_info "Available templates:"
+            pveam available -section system 2>/dev/null | grep -i debian | head -5
+        } >&2
         exit 1
     fi
 

